### PR TITLE
Implement simple InfluxDB client

### PR DESCRIPTION
**Pull request by wcoenraads**
Adds a simple InfluxDB client as described in #6. It's important to note that the current implementation does not escape the parameters to the query, so it should not be used with untrusted input.

### CODE REVIEW

File: src/influx.jl [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    password::String
+    InfluxClient(
+        host::String,
+        database::String,
+        port::Int = 8086,
+        path::String = "query",
+        username::String = "",
+        password::String = "",
+    ) = new(host, database, port, username, password, path)
+end
```
Comment (suvayu) [inline src/influx.jl:24]: I don't think this kind of convenience constructors should be inner.  You should move this to outer, then you cannot use `new`.  So, probably this:

```suggestion
end

InfluxClient(
    host::String,
    database::String,
) = InfluxClient(host, database, 8086, "query", "", "")
```

**Review by suvayu — CHANGES_REQUESTED** [review]
A few comments:
- Could you wrap the client in a module so that everything InfluxDB is in its own namespace?
  ```julia
  module Influx
  # contents of influx.jl
  end # module InfluxDB
  ```
- With the above change, maybe it is best to rename `InfluxClient` to just `Client`
- Do you have any idea how to write tests for this module?

**Comment by clizbe** [comment]
@wcoenraads Is this waiting on Suvayu's return?

**Comment by wcoenraads** [comment]
These comments/notifications completely passed me by - apologies! I'll address your comments and update the PR this afternoon.

As for testing, I'm also unsure. Downstream tests relying on these APIs could use mock data, but the influx client should really be tested against the InfluxDB API as that's the thing it interacts with. We could create a setup with a development DB to test against, but that feels like a lot of effort for little gain at the moment.

**Comment by wcoenraads** [comment]
>     * Could you wrap the client in a module so that everything InfluxDB is in its own namespace?
>       ```julia
>       module InfluxDB
>       # current contents of influx.jl
>       end # module InfluxDB
>       ```
> 
>     * With the above change, maybe it is best to rename `InfluxClient` to just `Client`

Good point on the module! 

I've renamed `InfluxClient` to `InfluxDBClient` to stay consistent with the [official API](https://influxdb-client.readthedocs.io/en/stable/api.html#influxdbclient). I'm also not super familiar with the conventional way to do imports in Julia, but I assume using `InfluxDBClient` instead of just `Client` should also help prevent conflicts when importing it.

**Comment by suvayu** [comment]
**PS:** I think the nightly test failure is due to an upstream packaging issue, ignore for now.

**Comment by wcoenraads** [comment]
> I think it would be more consistent if the file was called `influxdb.jl`.

Yep, good point - changed!

>  I can think of two approaches:
> [....]

The first approach could work, but seems limited and fragile: tests that can't be run in CI are much less likely to be adhered to, and they would depend on the specifics of an internal DB we only have limited control over.

The second approach is solid, but indeed more work. It might also be beneficial in tests for later components, where we can do end-to-end tests without mocks in between. But that's a concern for later.

**Comment by suvayu** [comment]
Thanks! I merged this, we can workout the tests in a separate PR.

**Comment by clizbe** [comment]
Closes #6

**Pull request merged**